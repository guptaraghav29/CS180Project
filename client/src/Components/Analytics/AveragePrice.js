import React, { useState, useEffect } from "react";

const AveragePrice = ({ carTableData, currentSelection }) => {
  const [averagePrice, setAveragePrice] = useState(0);

  useEffect(() => {
    let selectedChoices = currentSelection;
    let price = 0;
    carTableData.forEach((car) => {
      if (selectedChoices.includes(car.id)) {
        price += parseInt(car.price);
        selectedChoices = selectedChoices.filter((id) => id !== car.id);
      }
    });
    setAveragePrice(Math.floor(price / currentSelection.length));
  }, [carTableData, currentSelection]);

  return <div>Average Price: {averagePrice}</div>;
};

export default AveragePrice;
