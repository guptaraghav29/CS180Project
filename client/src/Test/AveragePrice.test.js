import * as React from "react";
import * as ReactDOM from "react-dom";
import AveragePrice from "../Components/Analytics/AveragePrice";
import renderer from "react-test-renderer";

import {
    fireEvent,
    getQueriesForElement,
    render,
} from "@testing-library/react";

it("testing the snapshot", () => {
	const tree = renderer.create(< AveragePrice />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("renders without any crashes", () => {
	const div = document.createElement("div");
	ReactDOM.render(< AveragePrice />, div);
});
