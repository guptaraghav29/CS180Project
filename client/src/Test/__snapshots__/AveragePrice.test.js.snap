// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the snapshot 1`] = `
<div>
  Average Price: 
  0
</div>
`;
